### PR TITLE
Add a MPA design and proto definition.

### PR DESCRIPTION
**Pull request by stvnrhodes**
I've prototyped this enough in https://github.com/stvnrhodes/sansshell/tree/mpa to have reasonable confidence that this design is implementable. I'm separating out the design and proto into an initial PR so that we can make sure that we agree on the high-level direction before we deeply review any implementation.

The readme is meant to be used as long-term documentation on how to use MPA.

Part of https://github.com/Snowflake-Labs/sansshell/issues/346

### CODE REVIEW

File: services/mpa/README.md
Comment (sfc-gh-ajoyner) [inline services/mpa/README.md:35]: nit: should probably say “as long as the command’s input remains the same and the sansshell-server approval is still in memory (the binary has not restarted, and it has not been cleared by a user)"

File: services/mpa/mpa.proto
Comment (sfc-gh-ajoyner) [inline services/mpa/mpa.proto:74]: It might be helpful for StoreRequest (and StoreResponse) to have a TTL.  Eventually you'll want to expire requests that weren't consumed, or it'll OOM sansshell-server in the limit conditions.  A client default of 72hrs or something probably puts an upper bound on the memory usage, without unnecessarily hampering typical usage.  WDYT?

Comment (sfc-gh-ajoyner) [reply services/mpa/mpa.proto:74]: Is there also a plan to cap the number of in-flight MPA requests that sansshell-server is willing to queue?  That would be another defense that might make the TTL unnecessary.

File: services/mpa/mpa.proto
Comment (sfc-gh-ajoyner) [inline services/mpa/mpa.proto:117]: This will sometimes contain sensitive data.  The (reasonable) advice from the README.md "It's highly recommended to additionally allow `/Mpa.Mpa/List`, `/Mpa.Mpa/Get`" runs the risk that you'll expose that sensitive data to a wide audience.  Complex policies could be written, but perhaps a simpler approach is for `StoreRequest` to allow the user to send a read ACL, which `Get` could enforce?  WDYT?

**Review by sfc-gh-ajoyner — APPROVED** [review]
The big picture looks right, so I'm going to go ahead and smash Approve.  I've left some comments we can discuss here, or that you can consider food-for-thought as the implementation develops.  None of them are show-stoppers, for sure.

File: services/mpa/mpa.proto
Comment (stvnrhodes) [reply services/mpa/mpa.proto:74]: I have a "Requests are stored in-memory and older requests may be cleared automatically" note in the RPC method comment that' should cover this.

I did "evict oldest request if there's >1000" in my prototype.  It's probably worth adding an "invalidate after X hours" so that old approvals don't cause confusion later on.

File: services/mpa/mpa.proto
Comment (stvnrhodes) [reply services/mpa/mpa.proto:117]: This part's a bit tricky. Get intentionally has nothing other than an ID in the request so rego policies can't do useful access control based on the request contents. I don't want to stick a read acl in StoreRequest for a couple reasons.

1. Users would have to specify more than `-mpa` when creating an MPA request
2. The access control is different than anything else we have and we'd either have a non-rego way of doing it or we'd have to have a weird rpcauth hook to allow doing it through rego

I've tweaked the wording a bit to say "let approvers do it" instead of "let anyone do it". In practice I think matching against a combination of peer identity and host information can give reasonable protection on who sees the data..

File: services/mpa/README.md
Comment (stvnrhodes) [reply services/mpa/README.md]: Good call, updated.

**Pull request merged**
